core/src/main/java/org/elasticsearch/threadpool/ThreadPoolStats.java

```diff
@@ -188,6 +206,7 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
         builder.startObject(Fields.THREAD_POOL);
+        Collections.sort(stats);
```

Comment:
I added it here because I just want the XContent output to be sorted. I think we don't really care otherwise and thus don't need to sort everytime we instantiate a ThreadPoolStats object. But I can move it if you really prefer.